src/services/notifier/handlers/provider.ts

```diff
@@ -34,6 +36,34 @@ export const handlers = {
     } else {
       logger.error(`Sequelize instance not found. Cannot update ${provider}'s plans.`)
     }
+  },
+  async SubscriptionCreated (event: SubscriptionCreated, { subscriptionService }: NotifierServices): Promise<void> {
+    const { provider, hash, consumer } = event.returnValues
+
+    const providerIns = await ProviderModel.findOne({ where: { provider } })
+
+    if (!providerIns) throw new Error(`Provider ${provider} not found`)
+
+    const [host, port] = providerIns.url.split(/(?::)(\d*)$/, 2)
+    const notifierService = new NotifierSvcProvider({ host, port })
+
+    const subscriptionResponse = await notifierService.getSubscriptions(consumer, [hash])
+
+    const [subscription] = subscriptionResponse
+    await SubscriptionModel.create({
+      hash,
+      providerId: provider,
+      consumer,
+      subscriptionId: subscription.id,
+      status: subscription.status,
+      subscriptionPlanId: subscription.subscriptionPlanId,
+      previousSubscription: subscription.previousSubscription,
+      expirationDate: new Date(subscription.expirationDate),
+      topics: subscription.topics
+    })
+
+    if (subscriptionService.emit) subscriptionService.emit('created', wrapEvent('SubscriptionCreated', { provider, consumer, hash }))
```

Comment:
Shouldn't the emitted object contain all of the data? That would be quite convenient for the client I'd say.